packages/cozy-konnector-libs/libs/signin.js

```diff
@@ -1,7 +1,15 @@
 /**
  * The goal of this function is to provide an handy method to log the user in,
- * on html form pages. On success, it resolves a promise with a parsed body. On
- * failure, it throws a `LOGIN_FAILED` error.
+ * on html form pages. On success, it resolves a promise with a parsed body.
+ *
+ * Errors:
+ *
+ * - LOGIN_FAILED if the validate predicate is false
+ * - INVALID_FORM if the element matched by `formSelector` is not a form or has
```

Comment:
I was wondering if I should throw a `LOGIN_FAILED` with an explicit log error message about the `form` not matching basic needs. WDYT ?
It would be more robust to layout changes by the provider.

Reply:
> It would be more robust to layout changes by the provider.

I don't understand, why "INVALID_FORM" would be more robust than "LOGIN_FAILED" ?

I think it is better like with "INVALID_FORM". LOGIN_FAILED would be interpreted by the connection form (the collect application) and a message would be displayed to the user saying that its credentials are invalid, and that would be wrong.

Reply:
I'm ok with your interpretation then.
I had the idea that LOGIN_FAILED would have a broader interpretation than wrong credentials.

Would we need to add the INVALID_FORM to the main errors documentation of the repo if we keep it like this?